deepmd/pt/train/training.py

```diff
@@ -1053,47 +1053,22 @@ def save_model(self, save_path, lr=0.0, step=0) -> None:
             checkpoint_files[0].unlink()
 
     def get_data(self, is_train=True, task_key="Default"):
-        if not self.multi_task:
-            if is_train:
-                try:
-                    batch_data = next(iter(self.training_data))
-                except StopIteration:
-                    # Refresh the status of the dataloader to start from a new epoch
-                    with torch.device("cpu"):
```

Comment:
You may not pass the tests if you remove this line.

Reply:
Will fix. But why this is only required for single task training data, not for multi task, not for valid data?

Reply:
Those lines might not be covered.

Reply:
I am not 100% sure about this comment. Please check the UT result...

Reply:
> I am not 100% sure about this comment. Please check the UT result...

haha you are right